tenable/ad/ldap_configuration/api.py

```diff
@@ -0,0 +1,76 @@
+'''
+LDAP Configuration
+=============
+
+Methods described in this section relate to the ldap configuration API.
+These methods can be accessed at ``TenableAD.ldap_configuration``.
+
+.. rst-class:: hide-signature
+.. autoclass:: LDAPConfigurationAPI
+    :members:
+'''
+from typing import Dict
+from tenable.ad.ldap_configuration.schema import LDAPConfigurationSchema
+from tenable.base.endpoint import APIEndpoint
+
+
+class LDAPConfigurationAPI(APIEndpoint):
+    _path = 'ldap-configuration'
+    _schema = LDAPConfigurationSchema()
+
+    def details(self) -> Dict:
+        '''
+        Get LDAP configuration singleton
+
+        Returns:
+            dict:
+                The LDAP configuration object
+
+        Examples:
+            >>> tad.ldap_configuration.details()
+        '''
+        return self._schema.load(self._get())
+
+    def update(self,
+               **kwargs
+               ) -> Dict:
+        '''
+        Update LDAP configuration singleton
+
+        Args:
+            enabled (optional, bool):
+                ???
+            url (optional, str):
+                ???
+            search_user_dn (optional, str):
+                ???
+            search_user_password (optional, str):
+                ???
+            user_search_base (optional, str):
+                ???
+            user_search_filter (optional, str):
+                ???
+            allowed_groups (optional, List[Dict]):
+                The below listed params are expected in allowed groups dict.
+            name (required, str):
+                ???
+            default_role_ids (required, List[int]):
+                ???
+            default_profile_id (required, int):
+                ???
```

Comment:
Ideally we should work out what these fields are and document them.

Reply:
We have updated param documentations.